CodeSniffer/Standards/Generic/Sniffs/Files/FileEncodingSniff.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Generic_Sniffs_Files_FileEncodingSniff.
+ *
+ * PHP version 5
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Klaus Purer <klaus.purer@mail.com>
+ * @copyright 2016 Klaus Purer
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+/**
+ * Generic_Sniffs_Files_FileEncodingSniff.
+ *
+ * Validates the encoding of a file against a white list of allowed encodings.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Klaus Purer <klaus.purer@mail.com>
+ * @copyright 2016 Klaus Purer
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ * @version   Release: @package_version@
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+class Generic_Sniffs_Files_FileEncodingSniff implements PHP_CodeSniffer_Sniff
+{
+
+    /**
+     * List of encodings that files may be encoded with.
+     *
+     * Any other detected encodings will throw a warning.
+     *
+     * @var array
+     */
+    public $allowedEncodings = array('UTF-8');
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        // Not all PHP installs have the multi byte extension - nothing we can do.
+        if (function_exists('mb_check_encoding') === false) {
+            return array();
+        }
```

Comment:
Since we are adding an optional dependency on the mbstring extension, it would be nice to add this as a suggestion to `composer.json`, so that users are made aware of it. For example:

```
{
    "suggest": {
        "ext-mbstring": "For checking file encodings"
    }
}
```

This will generate a message on installation:

> squizlabs/php-codesniffer suggests installing ext-mbstring (For checking file encodings)